src/bootstrap/src/core/build_steps/gcc.rs

```diff
@@ -34,17 +90,39 @@ pub enum GccBuildStatus {
     ShouldBuild(Meta),
 }
 
-/// This returns whether we've already previously built GCC.
+/// Tries to download GCC from CI if it is enabled and GCC artifacts
+/// are available for the given target.
+/// Returns a path to the libgccjit.so file.
+fn try_download_gcc(builder: &Builder<'_>, target: TargetSelection) -> Option<PathBuf> {
```

Comment:
IMO, it's better to keep this inside the `download` module along with the other components.

Also it's probably worth creating two functions with `cfg(test)` and `cfg(not(test)` similar to `maybe_download_ci_llvm`, so we don't end up downloading this while running unit tests with certain config combinations in the future.

Reply:
See https://github.com/rust-lang/rust/pull/138051#discussion_r1988945396 for why I don't necessarily want to do that.

Adding `cfg(test)` sounds reasonable, I'll add it.